### PR TITLE
fix: reenable floating point exceptions

### LINKED ISSUE #3281: Improved handling of NaNs
**What is the requested feature?**

Warning message when NaNs are found in GEOS, removing the ambiguity over the source of the issue (linear solver or GEOS)

**Is your request related to a specific problem?**

The following leads to a crash in hypre, but the problem is on the linear system assembled in GEOS.
```
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rflow ) = ( 2.18e-01 )        ( Rsolid ) = ( -nan )        ( R ) = ( -nan )
```

**Describe the solution you'd like**

A warning message like:
```
    Attempt:  0, ConfigurationIter:  0, NewtonIter:  0
        ( Rflow ) = ( 2.18e-01 )        ( Rsolid ) = ( -nan )        ( R ) = ( -nan )

*********************************************************************
*                    WARNING! WARNING! WARNING!                     *
*                                                                   *
*  The assembled sparse linear system contains invalid numbers      *
*  (NaNs). This may cause the solver to fail. Please check your     *
*  input values, boundary conditions, or model setup for potential  *
*  issues that might lead to these errors.                          *
*********************************************************************

```

### PR DESCRIPTION
**Pull request by rrsettgast**
This PR:

- [x] reenables floating point exceptions, which were disabled (accidentally??) by @tbeltzun in #2738 
- [x] ....and then fix the FPE's that occur in the code now that we are checking.

https://github.com/GEOS-DEV/LvArray/pull/327
https://github.com/GEOS-DEV/thirdPartyLibs/pull/279

resolves #3281

### CODE REVIEW

**Review by victorapm — APPROVED** [review]

**Review by paveltomin — APPROVED** [review]

**Review by CusiniM — APPROVED** [review]
src/coreComponents/constitutive/fluid/multifluid/CO2Brine/functions/CO2SolubilityDuanSun.cpp L68 also needs a small fix.

File: host-configs/LLNL/ruby-clang-14.cmake [hunk 1]
```diff
@@ -2,5 +2,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/../../src/coreComponents/LvArray/host-configs/
 
 # MPI
 set(MPI_HOME /usr/tce/packages/mvapich2/mvapich2-2.3.7-clang-14.0.6-magic CACHE PATH "")
+#set( CMAKE_CXX_FLAGS "-march=x86-64-v2 -mno-avx512f" CACHE STRING "" FORCE)
```
Comment (corbett5) [inline host-configs/LLNL/ruby-clang-14.cmake:6]: Do we want to keep these options commented out? If so maybe add a comment saying why they're there.

**Review by wrtobin — APPROVED** [review]

File: src/coreComponents/mainInterface/ProblemManager.cpp [hunk 2]
```diff
@@ -139,6 +139,8 @@ ProblemManager::ProblemManager( conduit::Node & root ):
     setRestartFlags( RestartFlags::WRITE ).
     setDescription( "Whether to disallow using pinned memory allocations for MPI communication buffers." );
 
+  chai::ArrayManager::getInstance()->disableCallbacks();
```
Comment (wrtobin) [inline src/coreComponents/mainInterface/ProblemManager.cpp:142]: why is this needed?

Comment (corbett5) [inline src/coreComponents/mainInterface/ProblemManager.cpp:142]: Won't this globally disable move logging? I thought this was a command line option?

**Review by corbett5 — APPROVED** [review]

File: src/coreComponents/mainInterface/ProblemManager.cpp (hunk 2 above)
Comment (rrsettgast) [reply src/coreComponents/mainInterface/ProblemManager.cpp:142]: I wanted this off by default for all the unitTests.

File: host-configs/LLNL/ruby-clang-14.cmake (hunk 1 above)
Comment (rrsettgast) [reply host-configs/LLNL/ruby-clang-14.cmake:6]: I can add a comment. Someone is going to have to come back and investigate.

File: src/coreComponents/mainInterface/ProblemManager.cpp (hunk 2 above)
Comment (rrsettgast) [reply src/coreComponents/mainInterface/ProblemManager.cpp:142]: It is...but the command line option is executed after this, but isn't called for any unitTests. Putting this here turns it off for the unit tests, as it is quite annoying for those. I assume it can be enabled after being disabled? Perhaps that is a bad assumption?

File: host-configs/LLNL/dane-gcc-12.cmake [hunk 3]
```diff
@@ -6,4 +6,6 @@ set(MPI_HOME /usr/tce/packages/mvapich2/mvapich2-2.3.7-gcc-12.1.1-magic CACHE PA
 # ATS
 set(ATS_ARGUMENTS "--machine slurm112"  CACHE STRING "")
 
+#set( CMAKE_CXX_FLAGS "-march=x86-64-v2 -mno-avx512f" CACHE STRING "" FORCE)
```
Comment (rrsettgast) [inline host-configs/LLNL/dane-gcc-12.cmake:10]:
```suggestion
# This is here to note the required flags for using valgrind. These will have to be propagated to the TPL's
#set( CMAKE_CXX_FLAGS "-march=x86-64-v2 -mno-avx512f" CACHE STRING "" FORCE)
```

File: host-configs/LLNL/ruby-clang-14.cmake (hunk 1 above)
Comment (rrsettgast) [inline host-configs/LLNL/ruby-clang-14.cmake:6]:
```suggestion
# This is here to note the required flags for using valgrind. These will have to be propagated to the TPL's
#set( CMAKE_CXX_FLAGS "-march=x86-64-v2 -mno-avx512f" CACHE STRING "" FORCE)
```

File: host-configs/LLNL/ruby-clang-14.cmake [hunk 4]
```diff
@@ -2,5 +2,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/../../src/coreComponents/LvArray/host-configs/
...
+#set( CMAKE_CXX_FLAGS "-g -O2 -fno-omit-frame-pointer -fsanitize=address" CACHE STRING "" FORCE)
```
Comment (rrsettgast) [inline host-configs/LLNL/ruby-clang-14.cmake:8]:
```suggestion
# This is what is required for using address sanitizer. This should be put into GeosxOptions.cmake when it is incorporated into the build options.
#set( CMAKE_CXX_FLAGS "-g -O2 -fno-omit-frame-pointer -fsanitize=address" CACHE STRING "" FORCE)
```

File: host-configs/LLNL/ruby-clang-14.cmake (hunk 1 above)
Comment (rrsettgast) [reply host-configs/LLNL/ruby-clang-14.cmake:6]: comments added.

File: src/coreComponents/mainInterface/ProblemManager.cpp (hunk 2 above)
Comment (corbett5) [reply src/coreComponents/mainInterface/ProblemManager.cpp:142]: I'm sure you're right as you mentioned in the other comment that it can be re-enabled if desired.

**Comment by paveltomin** [comment]
CC @drmichaeltcvx
